lib/stat_tracker.rb

```diff
@@ -98,6 +98,34 @@ def count_of_teams
     @team_collection.teams.length
   end
 
+  def best_offense
+    game_teams_grouped_by_team_id = @gtc.game_teams.group_by do |game_team|
+      game_team.team_id
+    end
+    x = game_teams_grouped_by_team_id.each_pair do |team_id, games_by_team|
+      total_goals = games_by_team.map do |single_game|
+        single_game.goals
+      end
+      game_team_averages = (total_goals.sum.to_f / total_goals.length).round(2)
+      (game_teams_grouped_by_team_id[team_id] = game_team_averages)
+    end
+    team_name_by_id(x.key(x.values.max))
+  end
+
+  def worst_offense
+    game_teams_grouped_by_team_id = @gtc.game_teams.group_by do |game_team|
+      game_team.team_id
+    end
+    x = game_teams_grouped_by_team_id.each_pair do |team_id, games_by_team|
+      total_goals = games_by_team.map do |single_game|
+        single_game.goals
+      end
+      game_team_averages = (total_goals.sum.to_f / total_goals.length).round(2)
+      (game_teams_grouped_by_team_id[team_id] = game_team_averages)
+    end
+    team_name_by_id(x.key(x.values.min))
```

Comment:
This is a handy way of calling the hash values, will be using on my method refactor